library/alloc/src/raw_vec.rs

```diff
@@ -77,6 +82,237 @@ pub(crate) struct RawVec<T, A: Allocator = Global> {
     alloc: A,
 }
 
+#[inline]
+const fn max_size_for_align(align: usize) -> usize {
```

Comment:
Yeah there's a lot of code reuse issues with the PR right now. This one is particularly glaring; though I'm not sold on using `Alignment` because we can either pass in `Alignment::of::<T>()` instead of `align_of::<T>()` but that's extra MIR on the polymorphic side, or we can convert our `usize` into an `Alignment` only to _immediately_ convert it back inside of `Layout::max_size_for_align`. 🤷 Maybe I'll be more sympathetic to the latter approach if this looks good in perf and I get to polishing.